dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguageHandlerInterface.java

```diff
@@ -960,6 +957,17 @@ default Optional<String> convertMapsToContent(final String mainDescName, final T
         return Optional.empty();
     }
 
+    /**
+     * Indicates if the workflow version has a test parameter file with all file inputs values
+     * being open data.
+     * @param workflowVersion
+     * @param checkUrlLambdaUrl
+     * @return if unable to determine, Optional.empty(), otherwise an non-empty boolean
+     */
+    default Optional<Boolean> hasPublicAccessibleUrls(WorkflowVersion workflowVersion, final String checkUrlLambdaUrl) {
```

Comment:
Idea: instead of passing in the lambda checking url, which couples the ensuing code to that particular mechanism, we could create a `UrlChecker` interface, and create an implementation that uses the lambda, and pass that to `hasPublicAccessibleUrls`.  Something like:
```
interface UrlChecker {
    Optional<Boolean> checkUrls(Set<String> urls);
}

public class LambdaUrlChecker implements UrlChecker {
    private final String lambdaUrl;
    public LambdaUrlChecker(String lambdaUrl) {
        this.lambdaUrl = lambdaUrl;
    }
    Optional<Boolean> checkUrls(Set<String> urls) {
        return CheckUrlHelper.checkUrls(urls, lambdaUrl);
    }
}
```
That way, if we ever change the url checking mechanism, like move it to something other than the lambda, it's super easy, and the existing code just works.  We can also mock the url checking more easily.

In general, for future refactors, I recommend we do more of this kind of thing.

Reply:
Good idea, added.